lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/IMRUBroadcastReduceWithFilePartitionDataSetTest.cs

```diff
@@ -0,0 +1,360 @@
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using Org.Apache.REEF.Common.Tasks;
+using Org.Apache.REEF.IMRU.API;
+using Org.Apache.REEF.IMRU.OnREEF.Driver;
+using Org.Apache.REEF.IO.PartitionedData;
+using Org.Apache.REEF.IO.PartitionedData.FileSystem;
+using Org.Apache.REEF.IO.TempFileCreation;
+using Org.Apache.REEF.Network.Examples.GroupCommunication.BroadcastReduceDriverAndTasks;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Formats;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Utilities.Diagnostics;
+using Org.Apache.REEF.Utilities.Logging;
+using Xunit;
+using TraceLevel = System.Diagnostics.TraceLevel;
+
+namespace Org.Apache.REEF.Tests.Functional.IMRU
+{
+    /// <summary>
+    /// This class tests DataLoadingContext for IMRU with FilePartitionDataSet
+    /// </summary>
+    [Collection("FunctionalTests")]
+    public class IMRUBroadcastReduceWithFilePartitionDataSetTest : IMRUBrodcastReduceTestBase
+    {
+        /// <summary>
+        /// Defines the number of data in input data bytes in the test
+        /// </summary>
+        private const int DataCount = 3;
+
+        /// <summary>
+        /// This test tests DataLoadingContext with FilePartitionDataSet
+        /// The IPartitionedInputDataSet configured in this test is passed to IMRUDriver, and IInputPartition associated with it is injected in 
+        /// DataLoadingContext for the Evaluator. Cache method in IInputPartition is called when the context starts. 
+        /// Mapper function injected in this test will verify if the temp file downloaded exists.
+        /// </summary>
+        [Fact]
+        public void TestWithFilePartitionDataSetOnLocalRuntime()
+        {
+            int chunkSize = 2;
+            int dims = 10;
+            int iterations = 10;
+            int mapperMemory = 5120;
+            int updateTaskMemory = 5120;
+            int numTasks = 4;
+            string testFolder = DefaultRuntimeFolder + TestId;
+            TestBroadCastAndReduce(false,
+                numTasks,
+                chunkSize,
+                dims,
+                iterations,
+                mapperMemory,
+                updateTaskMemory,
+                testFolder);
+            ValidateSuccessForLocalRuntime(numTasks, 0, 0, testFolder);
```

Comment:
For verification we rely on the fact that if files are not copied, the tasks will fail to be constructed, and we'll get non-zero number of failed tasks, right? Might be worth to leave a comment about this in code, since for IMRU tests we tend to to more validation than just `ValidateSuccessForLocalRuntime`, so having no extra validation is a bit confusing